### PR TITLE
EVEREST-107: tool to automate veneer updates

### PR DESCRIPTION
**Pull request by michal-kralik**
A tool to automate veneer file updates.

**Usage**
Create a new patch version:
```
go run ./tools/ \
  --veneer-file veneer/everest-operator.yaml \
  --version-type patch \
  --channel fast-v0 \
  --new-version 0.9.2 \
  --current-version 0.9.1
```

Create a new minor version:
```
go run ./tools/ \
  --veneer-file veneer/everest-operator.yaml \
  --version-type minor \
  --channel fast-v0 \
  --new-version 0.10.0 \
  --current-version 0.9.1
```

### CODE REVIEW

**Comment by oksana-grishchenko** [comment]
Thanks Michal for implementing the tool!

What would be the usage for the RCs? 
There are the following cases that might be handled slightly different:

1. An RC of a new minor version: 
   latest version: `0.9.0`, new version `0.10.0-rc1` 
2. Next RC for the same minor or patch version:
   latest version: `0.10.0-rc1`, new version `0.10.0-rc2`  or
   latest version: `0.10.1-rc1`, new version `0.10.1-rc2`
3. An RC of a new patch. version:
   latest version: `0.10.0`, new version `0.10.1-rc1` 

All the RC versions are applied only to the fast channel.

**Comment by michal-kralik** [comment]
This shall be all covered by the existing functionality.
There's no distinction between rc and non-rc.

I added a couple more tests to make sure we have this covered as well.

**Review by oksana-grishchenko — APPROVED** [review]
Thanks for adding the tests for the RCs. The code looks good to me, great job, I'm glad we're going to have the veneer update automated❤️ 

So for the RCs the release CI is going to run the tool once (for the fast channel)
For the releases the release CI is going to run the tool twice (for the fast and the stable channel). And we can use smth like https://github.com/marketplace/actions/get-latest-release to get the `current-version`

File: tools/veneer-update_test.go [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+    - name: everest-operator.v0.9.1
+      skips:
+        - everest-operator.v0.9.0
+name: stable-v0
+package: everest-operator
+schema: olm.channel
+---
+image: docker.io/perconalab/everest-operator-bundle:0.0.0
+schema: olm.bundle
+---
```
Comment (recharte) [inline tools/veneer-update_test.go:135]: Aren't we missing adding the bundle of the newly added version?

**Review by recharte — COMMENTED** [review]
Great job Michal, the mutation of the `olm.channel` looks great but I think we are missing adding the `olm.bundle` of the new version.

File: tools/veneer-update_test.go (hunk 1 above)
Comment (michal-kralik) [reply tools/veneer-update_test.go:135]: Yes, we were. Fixed now.

**Review by recharte — APPROVED** [review]

**Pull request merged**